docs/tutorials/paroutes_benchmark.ipynb

```diff
@@ -0,0 +1,749 @@
+      "text/plain": [
+       "<rdkit.Chem.rdchem.Mol at 0x7f3b14233ba0>"
+      ]
+     },
+     "execution_count": 19,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "target_mol = Molecule(test_smiles[100])\n",
+    "target_mol.rdkit_mol"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 20,
+   "id": "5af33f0f-daad-49ea-8d36-2d898fec07d6",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "CPU times: user 5.2 s, sys: 17.4 ms, total: 5.22 s\n",
+      "Wall time: 5.21 s\n"
+     ]
+    }
+   ],
+   "source": [
+    "%%time\n",
+    "# Run retro-star\n",
+    "retro_star_alg.reset()\n",
+    "retro_star_search_graph, _ = retro_star_alg.run_from_mol(target_mol)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 21,
+   "id": "d6af0d40-c7fb-48d8-b63c-42762e46976d",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "CPU times: user 5.26 s, sys: 14 ms, total: 5.28 s\n",
+      "Wall time: 5.23 s\n"
+     ]
+    }
+   ],
+   "source": [
+    "%%time\n",
+    "# Run MCTS\n",
+    "mcts_alg.reset()\n",
+    "mcts_search_graph, _ = mcts_alg.run_from_mol(target_mol)"
+   ]
```

Comment:
I wasn't calling `reset` before running the search algorithm in the other tutorials. Is it always necessary to do so? If yes, why is it not done automatically in `__init__`?

Reply:
It is _not_ necessary (and in production you would probably not call it). I called it here for two reasons:

1. This tutorial is (sort of) comparing MCTS and retro*, but if I run MCTS after retro*, the reaction model will still have a lot of reactions in its cache, effectively giving MCTS a headstart in its search.
2. The parameter `limit_reaction_model_calls` gives an _absolute_ limit on the number of calls, not a limit relative to the start of the algorithm. By resetting it, I ensure that both algorithms get the same number of calls. To be fair though, this behaviour is a bit unintuitive and I would be happy to change it.

Reply:
Ah, I see. The "not giving a head start to the second algorithm" point is quite nuanced, maybe even worth mentioning somewhere in the tutorial?

Reply:
Good point: I just pushed a commit with a more detailed explanation for this.